src/index.js

```diff
@@ -432,6 +432,36 @@ class MiniCssExtractPlugin {
           return source;
         }
       );
+
+      compilation.hooks.optimizeChunks.tap(pluginName, (chunks) => {
+        for (const chunk of chunks) {
+          chunk.modulesIterable.forEach((module) => {
+            // eslint-disable-next-line no-param-reassign
+            module.dependencies = module.dependencies.filter((dep) => {
+              const hasCssDep =
+                dep.module &&
+                dep.module.dependencies.find(
+                  (d) => d.module && d.module.type === MODULE_TYPE
+                );
+
+              if (hasCssDep) {
+                dep.disconnect();
```

Comment:
I think this works only for ESM dependencies. `require()` probably break because of this.

Could you add a CommonJS test cases?

Reply:
@sokra this is definitely will break CJS dependencies. Do you have any thoughts on how to avoid that?